src/os/macosx/string_osx.cpp

```diff
@@ -306,6 +306,9 @@ int MacOSStringCompare(const char *s1, const char *s2)
 	CFStringRef cf1 = CFStringCreateWithCString(kCFAllocatorDefault, s1, kCFStringEncodingUTF8);
 	CFStringRef cf2 = CFStringCreateWithCString(kCFAllocatorDefault, s2, kCFStringEncodingUTF8);
 
+	/* Don't know why add  '+ 2', but it's added because it already existed in the final function's return */
+	if (cf1 == nullptr or cf2 == nullptr) return (int)kCFCompareEqualTo + 2;
```

Comment:
Use the `||` operator instead of `or` as keyword (does that even compile?)
```suggestion
	if (cf1 == nullptr || cf2 == nullptr) return (int)kCFCompareEqualTo + 2;
```
I don't like that the +2 is added for unknown reasons, if it's necessary it needs an explanation for why.

Reply:
One would only have to look at the function comment to know why the return values are what they are:
`@return 1 if s1 < s2, 2 if s1 == s2, 3 if s1 > s2, or 0 if not supported by the OS.`

Having said that, I'd say that a return value of 0 would be the correct return. It will make OpenTTD fall back to a simple string compare, which is not as bad as just pretending the strings are equal.

Reply:
Oh, you both are right. I was tired of debugging (was late) when I made the code change, after I discovered where was the segfault, and I didn't realise on the function comments neither I was writing pythonic "or" style :(